ping/go/main.go

```diff
@@ -13,13 +13,15 @@ import (
 
 	"github.com/libp2p/go-libp2p-core/peer"
 
-	"github.com/libp2p/go-libp2p-noise"
+	noise "github.com/libp2p/go-libp2p-noise"
```

Comment:
This repo is deprecated as of v0.20.0. Same for -tls.

To do things the correct way, we should define a `NoiseProtocolID` in `compat`.

Reply:
👍

It looks like we also need to pass the Protocol constructor (`security(noise.ID, noise.New)`), I've implemented a getByName, then moved it into the compat package (less confident about this one). 

- https://github.com/libp2p/test-plans/pull/23/commits/907546613f25a989a35a3f5ef1e382157c487d87
- https://github.com/libp2p/test-plans/pull/23/commits/0f506ecda1042b76b2c722f7fa73d29ee6db7ec6